testsuite/MDAnalysisTests/analysis/test_bat.py

```diff
@@ -127,3 +127,8 @@ def test_bat_incorrect_dims(self, bat_npz):
         errmsg = 'Dimensions of array in loaded file'
         with pytest.raises(ValueError, match=errmsg):
             R = BAT(selected_residues, filename=bat_npz)
+
+    def test_bat_input_data(self,torsions,bat_frame):
+        arr1 = np.array(torsions)
+        arr2 = np.array(bat_frame)
+        return np.may_share_memory(arr1,arr2)
```

Comment:
The test does not directly test the problem described in #3501 . 

I would look at the original issue report and see if I can turn the report into a test. Your test should _fail_ with the code before your fix and pass with your fix.